src/context.rs

```diff
@@ -235,6 +235,10 @@ impl<'a> Context<'a> {
         disabled == Some(true)
     }
 
+    pub fn detect_env_vars(&'a self, env_vars: &'a [&'a str]) -> bool {
+        env_vars.is_empty() || (env_vars.iter().any(|e| self.get_env(e).is_some()))
```

Comment:
I think this signature will work better, to allow use with `?`:
```suggestion
    pub fn detect_env_vars(&'a self, env_vars: &'a [&'a str]) -> Option<()> {
        (env_vars.is_empty() || (env_vars.iter().any(|e| self.get_env(e).is_some()))).then_some(())
```